admin/main.js

```diff
@@ -1,85 +1,76 @@
 function main() {
-  const username = document.cookie.split('=')[1];
+  // see https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie
+  const username = document.cookie.replace(/(?:(?:^|.*;\s*)username\s*\=\s*([^;]*).*$)|^.*$/, "$1");
   const list = document.getElementById('user-list');
   const addButton = document.getElementById('hint-add-user');
-  // const goHomeButtom = document.getElementById('go-home');
-  addButton.addEventListener("click", () => {
-    document.querySelector('.add-user').hidden = !document.querySelector('.add-user').hidden
-  })
-  // addButton.addEventListener("click", () => {
-    
-  // })
+  let users = [];
+
+  addButton.addEventListener('click', () => {
+    const addUserWidget = document.querySelector('.add-user');
+    addUserWidget.hidden = !addUserWidget.hidden;
+  });
   const logoutButton = document.getElementById('logout')
-  const loginButton = document.getElementById('login');
+  const addUserButton = document.getElementById('add-user-button');
 
   function postData(url = '', data = {}) {
-      fetch(url, {
-          method: 'POST',
-          headers: {
-            'Content-Type': 'application/json'
-          },
-          body: JSON.stringify(data)
-      }).then(response => {
-        response.json()
-        if (response.ok) {
-          location.reload(true);
-        };
-        console.log(response)
-        return response
-      } )     //  return response.json();
+    return fetch(url, {
```

Comment:
I simplified this. This function should not act if the post fails. The caller should handle that in different ways.